testsuite/MDAnalysisTests/formats/test_libmdaxdr.py

```diff
@@ -21,376 +21,485 @@
 #
 from __future__ import print_function, absolute_import
 
-from nose.tools import raises
-from numpy.testing import assert_equal, assert_array_equal
-from numpy.testing import assert_array_almost_equal
+import numpy as np
+
 from numpy.testing import assert_almost_equal
+from numpy.testing import assert_array_almost_equal
+from numpy.testing import assert_array_equal
 
 from MDAnalysis.lib.formats.libmdaxdr import TRRFile
-from MDAnalysisTests.datafiles import TRR_single_frame
 from MDAnalysisTests.datafiles import TRR_multi_frame
+from MDAnalysisTests.datafiles import TRR_single_frame
 
 from MDAnalysis.lib.formats.libmdaxdr import XTCFile
-from MDAnalysisTests.datafiles import XTC_single_frame
 from MDAnalysisTests.datafiles import XTC_multi_frame
+from MDAnalysisTests.datafiles import XTC_single_frame
 
-from MDAnalysisTests.tempdir import run_in_tempdir
-import numpy as np
+import pytest
+
+XTC_OFFSETS = np.array([0, 104, 208, 312, 416, 520, 624, 728, 832, 936])
+TRR_OFFSETS = np.array([0, 480, 960, 1440, 1920, 2400, 2880, 3360, 3840, 4320])
+
+
+def test_xtc():
+    f = XTCFile(XTC_single_frame)
+    assert f.n_atoms == 10
+    xyz, box, step, time, prec = f.read()
+    # xtc only saves with 3 decimal places precision
+    assert_array_almost_equal(xyz.flat, np.arange(30), decimal=3)
+    assert_array_almost_equal(box, np.eye(3) * 20, decimal=3)
+    assert step == 0
+    assert time == 10.0
+    assert prec == 1000.0
+
+
+def test_trr():
+    f = TRRFile(TRR_single_frame)
+    assert f.n_atoms == 10
+    frame = f.read()
+    # trr only saves with 3 decimal places precision
+    assert_array_almost_equal(frame.x.flat, np.arange(30))
+    assert_array_almost_equal(frame.v.flat, np.arange(30))
+    assert_array_almost_equal(frame.f.flat, np.arange(30))
+    assert_array_almost_equal(frame.box, np.eye(3) * 20)
+    assert frame.step == 10
+    assert frame.time == 5.0
+    assert frame.lmbda == 0.5
 
 
-class XDRFormatBaseTest(object):
```

Comment:
You could have kept the class and parametrised it - do you prefer the standalone functions instead?